bundles/net.tourbook/src/net/tourbook/weather/openweathermap/OpenWeatherMapRetriever.java

```diff
@@ -268,14 +325,27 @@ private TimeMachineResult deserializeWeatherData(final String weatherDataRespons
       } catch (final Exception e) {
 
          StatusUtil.logError(
-               "OpenWeatherMapRetriever.deserializeWeatherData : Error while " + //$NON-NLS-1$
+               "OpenWeatherMapRetriever.deserializeHistoricalWeatherData : Error while " + //$NON-NLS-1$
                      "deserializing the historical weather JSON object :" //$NON-NLS-1$
                      + weatherDataResponse + UI.SYSTEM_NEW_LINE + e.getMessage());
       }
 
       return newTimeMachineResult;
    }
 
+   private int getRequestCount() {
+
+      int requestCount;
+      try {
+
+         requestCount = _state.getInt(STATE_REQUEST_COUNT);
+
+      } catch (final NumberFormatException e) {
+         requestCount = 0;
+      }
```

Comment:
For getting state values you should use the `net.tourbook.common.util.Util.getState...()` methods

Reply:
Good to know, those functions are useful. 
Fixed